### PR TITLE
Issue4238 - Image build runc

### PR DESCRIPTION
**Pull request by SuryanarayanaPeri**
This is to add the metadata labels to rke2 images per https://github.com/rancher/rke2/issues/4238 for runc CLI

### CODE REVIEW

**Comment by SuryanarayanaPeri** [comment]
Please review and approve this PR

**Comment by brandond** [comment]
I'm not sure how useful this is. We don't actually ship this runc image anywhere, the only place it's used is to copy the built binary into the rke2-runtime image.

https://github.com/rancher/rke2/blob/v1.31.1%2Brke2r1/Dockerfile#L123-L128

**Comment by SuryanarayanaPeri** [comment]
> I'm not sure how useful this is. We don't actually ship this runc image anywhere, the only place it's used is to copy the built binary into the rke2-runtime image.
> 
> https://github.com/rancher/rke2/blob/v1.31.1%2Brke2r1/Dockerfile#L123-L128

@brandond - From my understanding, this is for the standardization across all the images - hence this PR to apply the labels.

File: Dockerfile [hunk 1]
```diff
@@ -28,4 +28,8 @@ RUN install -s runc /usr/local/bin
 RUN runc --version
 
 FROM scratch
+ARG GO_IMAGE
+LABEL org.opencontainers.image.url="https://hub.docker.com/r/rancher/runc"
```
Comment (brandond) [inline Dockerfile:32]:
```suggestion
LABEL org.opencontainers.image.url="https://hub.docker.com/r/rancher/hardened-runc"
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -28,4 +28,8 @@ RUN install -s runc /usr/local/bin
 RUN runc --version
 
 FROM scratch
+ARG GO_IMAGE
+LABEL org.opencontainers.image.url="https://hub.docker.com/r/rancher/hardened-runc"
+LABEL org.opencontainers.image.source="https://github.com/rancher/image-build-runc"
+LABEL org.opencontainers.image.base.name="${GO_IMAGE}"
```
Comment (brandond) [inline Dockerfile:34]: is this technically correct? This is a `FROM scratch` image with a single binary in it. Scratch images do not have a base image.

Comment (SuryanarayanaPeri) [reply Dockerfile]: Made the changes; Please review

Comment (SuryanarayanaPeri) [reply Dockerfile]: @brandond - Can you please review the latest changes applied and let me know your comments.

**Review by brandond — APPROVED** [review]

**Comment by SuryanarayanaPeri** [comment]
@brandond - thank you for the approval; I see 1 check failing due to trivy on amd64 - Any suggestions - I have been looking at the github pages related to trivy discussions and seeking your inputs.

**Pull request merged**